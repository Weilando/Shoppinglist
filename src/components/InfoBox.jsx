import React from 'react';
import PropTypes from 'prop-types';

function InfoBox(props) {
  return (
    <div className="InfoBox">
      <p>Entries: {props.entryCount} &emsp; Done entries: {props.doneEntryCount}</p>
    </div>
  );
}

InfoBox.propTypes = {
  entryCount: PropTypes.number.isRequired,
  doneEntryCount: PropTypes.number.isRequired,
};

InfoBox.defaultProps = {
  entryCount: 0,
  doneEntryCount: 0,
}

export default InfoBox;
