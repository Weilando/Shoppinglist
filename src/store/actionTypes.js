export const ADD_ENTRY = 'ADD_ENTRY';
export const UPDATE_ENTRY = 'UPDATE_ENTRY';
export const DELETE_ENTRY = 'DELETE_ENTRY';
export const TOGGLE_STATUS = 'TOGGLE_STATUS';

export const UPDATE_MEAL_SUGGESTIONS = 'UPDATE_MEAL_SUGGESTIONS';
export const REMOVE_DUPLICATES = 'REMOVE_DUPLICATES';
