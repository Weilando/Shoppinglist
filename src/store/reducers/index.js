import { generalReducer } from './generalReducer';
import { mealSuggestionsReducer } from './mealSuggestionsReducer';
import { combineReducers } from 'redux';

// rootReducer
export default combineReducers({
    general: generalReducer,
    mealSuggestions: mealSuggestionsReducer
});
