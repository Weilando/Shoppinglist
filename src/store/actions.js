import { ADD_ENTRY, UPDATE_ENTRY, DELETE_ENTRY, TOGGLE_STATUS, UPDATE_MEAL_SUGGESTIONS, REMOVE_DUPLICATES } from './actionTypes';

let nextEntryId = 0;

export function addEntry(newContent) {
  return {
    type: ADD_ENTRY,
    payload: {
      id: ++nextEntryId,
      newContent
    }
  }
}

export function updateEntry(id, newContent) {
  return {
    type: UPDATE_ENTRY,
    payload: {
      id: id,
      newContent
    }
  }
}

export function deleteEntry(id) {
  return {
    type: DELETE_ENTRY,
    payload: { id: id }
  }
}

export function toggleStatus(id) {
  return {
    type: TOGGLE_STATUS,
    payload: { id: id }
  }
}

export function updateMealSuggestions(mealSuggestions) {
  return {
    type: UPDATE_MEAL_SUGGESTIONS,
    payload: { newMealSuggestions: mealSuggestions }
  }
}

export function removeDuplicates() {
  return {
    type: REMOVE_DUPLICATES
  }
}
