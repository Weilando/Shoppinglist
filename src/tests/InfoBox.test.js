import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/store';
import InfoBox from '../components/InfoBox';

describe('snapshot tests for InfoBox component', () => {
  it('should render InfoBox component correctly', () => {
    const toggleFunction = jest.fn();
    const tree = renderer.create(
      <Provider store={store}>
        <InfoBox
          entryCount={42}
          doneEntryCount={7}
        />
      </Provider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  })
});
