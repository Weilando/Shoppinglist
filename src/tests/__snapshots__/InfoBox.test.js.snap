// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot tests for InfoBox component should render InfoBox component correctly 1`] = `
<div
  className="InfoBox"
>
  <p>
    Entries: 
    42
     â€ƒ Done entries: 
    7
  </p>
</div>
`;
